.github/actions/e2e-testing/edge/action.yml

```diff
@@ -27,27 +31,68 @@ runs:
       uses: GuillaumeFalourd/clone-github-repo-action@main
       with:
         owner: 'pinecone-io'
-        repository: 'pinecone-rag-demo'
+        repository: 'ts-client-e2e-tests'
         branch: 'main'
 
-    - name: Move package to pinecone-rag-demo
-      run: cd pinecone-rag-demo && mv ../pinecone-database-pinecone-*.tgz .
+    - name: Move package to ts-client-e2e-tests
+      run: cd ts-client-e2e-tests && mv ../pinecone-database-pinecone-*.tgz .
       shell: bash
 
-    - name: Install rag-demo deps
-      run: cd pinecone-rag-demo && npm install
+    - name: Install e2e tests dependencies
+      run: cd ts-client-e2e-tests && npm install
       shell: bash
 
     - name: Install pinecone-ts-client "main" branch code into the Vercel app
-      run: cd pinecone-rag-demo && npm install pinecone-database-pinecone-*.tgz
+      run: cd ts-client-e2e-tests && npm install pinecone-database-pinecone-*.tgz
       shell: bash
 
     - name: Install Vercel CLI
-      run: cd pinecone-rag-demo && npm install vercel
+      run: cd ts-client-e2e-tests && npm install vercel
       shell: bash
 
-    - name: Deploy Project Artifacts to Vercel
+    - name: Send POST request to endpoint Vercel app endpoint api/createSeedQuery
       run: |
-        cd pinecone-rag-demo
-        vercel --scope=pinecone-io --token=${{ inputs.vercel-token }} --yes
+        API_URL="https://ts-client-e2e-tests.vercel.app/api/createSeedQuery"
+        API_KEY="${{ inputs.PINECONE_API_KEY }}"
+
+        response=$(curl --silent --location --request POST "$API_URL" --header "PINECONE_API_KEY: $API_KEY")
+
+        indexName=$(echo "$response" | jq -r '.indexName')
```

Comment:
Index names are randomized in the app. We then save the index name to the GITHUB_ENV environment variable file, so that we can delete them later -- this ensures we can have simultaneous runs of this workflow, from creating the index to deleting it later

Reply:
Would it be possible to port the logic for the actual testing to an explicit TypeScript / Javascript file that handles the calling / asserting, etc?

Something similar to `IntegrationRunner.js`. I think this will generally be easier to build out and manage if it lives in a code-based script rather than in a GH actions file.

Reply:
Definitely! Done. Now we have a file called `src/end-to-end/localRuns.ts`